llvm/lib/Transforms/IPO/MergeFunctions.cpp

```diff
@@ -375,9 +375,35 @@ bool MergeFunctions::doFunctionalCheck(std::vector<WeakTrackingVH> &Worklist) {
 }
 #endif
 
+// This function iterates over two functions and compares their
+// metadata to make sure we aren't merging functions which have
+// distinct metadata that is different. We don't care about
+// debug data here, it shouldn't change MIR like CFI would.
```

Comment:
I think the "debug" thing should be commented only in the function, if at all. Note that one of debug info's primary design goals is to avoid affecting the results of optimizations (should get identical code with or without `-g`)—that's why the `instructionsWithDebug()` API exists—so it might be evident enough without a comment.

For CFI, it might be worth having a full sentence explaining the motivating llvm.type.test example? I'd also be fine with not mentioning CFI here.